### ISSUE TITLE
docker.running (and docker.build) actually do things with test=True

### ISSUE THREAD

**Issue by arthurlogilab** [saltstack/salt#18508]
When running a docker.build state with test=True it seems that salt is actually trying to build the image instead of just doing a dry-run. 

```
# salt-call state.highstate test=True
[snip]
[INFO    ] Running state [logilab-registry] at time 12:26:25.334501
[INFO    ] Executing state docker.built for logilab-registry
[INFO    ] Client {}
[INFO    ] Client._timeout= <type 'int'>
[INFO    ] Client {}
[INFO    ] Client._timeout= <type 'int'>
Ctrl-C
# salt-call --versions-report
           Salt: 2014.1.13
         Python: 2.7.8 (default, Oct 18 2014, 12:50:18)
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
         PyYAML: 3.11
          PyZMQ: 14.4.0
            ZMQ: 4.0.5
```

**Comment by martin-helmich** [comment]
I'm having the same issue with `docker.pulled`.

**Comment by rallytime** [comment]
Thanks for the update @martin-helmich. Can you post the output of `salt --versions-report`? The original report was against 2014.1.13, but I am wondering what version you are running on as well.

**Comment by martin-helmich** [comment]
Thanks for the quick response, @rallytime. Sure, here you go:

```
           Salt: 2014.7.0
         Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
 msgpack-python: 0.1.10
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 13.1.0
           RAET: Not Installed
            ZMQ: 3.2.3
           Mako: 0.7.0
```

**Comment by rallytime** [comment]
Thanks!

**Comment by stanislavb** [comment]
Replicating this behavior on Amazon Linux 2015.03. test=True actually does the operations.

```
           Salt: 2015.5.0
         Python: 2.6.9 (unknown, Apr  1 2015, 18:16:00)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
```

**Comment by rallytime** [comment]
Thanks for the extra info @stanislavb

**Comment by rallytime** [comment]
This was resolved with #25126 in the develop branch, which has been backported to 2015.8 with #25184. The fix is available in 2015.8.0. I am going to close this, but if any problems remain or pop up here, leave a comment and we will gladly reopen.
